cmake/OpenCVGenPkgconfig.cmake

```diff
@@ -8,71 +8,82 @@
 #
 # ${BIN_DIR}/unix-install/opencv.pc -> For use *with* "make install"
 # -------------------------------------------------------------------------------------------
-set(prefix      "${CMAKE_INSTALL_PREFIX}")
-set(exec_prefix "\${prefix}")
-set(libdir      "") #TODO: need link paths for OpenCV_EXTRA_COMPONENTS
-set(includedir  "\${prefix}/${OPENCV_INCLUDE_INSTALL_PATH}")
-
-if(CMAKE_BUILD_TYPE MATCHES "Release")
-  set(ocv_optkind OPT)
-else()
-  set(ocv_optkind DBG)
-endif()
 
-#build the list of opencv libs and dependencies for all modules
-set(OpenCV_LIB_COMPONENTS "")
-set(OpenCV_EXTRA_COMPONENTS "")
-foreach(m ${OPENCV_MODULES_PUBLIC})
-  list(INSERT OpenCV_LIB_COMPONENTS 0 ${${m}_MODULE_DEPS_${ocv_optkind}} ${m})
-  if(${m}_EXTRA_DEPS_${ocv_optkind})
-    list(INSERT OpenCV_EXTRA_COMPONENTS 0 ${${m}_EXTRA_DEPS_${ocv_optkind}})
-  endif()
-endforeach()
+macro(fix_prefix lst isown)
+  set(_lst)
+  foreach(item ${${lst}})
+    if(TARGET ${item})
+      get_target_property(item "${item}" LOCATION_${CMAKE_BUILD_TYPE})
+      if("${isown}")
+        get_filename_component(item "${item}" NAME_WE)
+        string(REGEX REPLACE "^lib(.*)" "\\1" item "${item}")
+      endif()
+    endif()
+    if(item MATCHES "^-l")
+      list(APPEND _lst "${item}")
+    elseif(item MATCHES "[\\/]")
+      get_filename_component(libdir "${item}" PATH)
+      get_filename_component(libname "${item}" NAME_WE)
+      string(REGEX REPLACE "^lib(.*)" "\\1" libname "${libname}")
+      list(APPEND _lst "-L${libdir}" "-l${libname}")
+    else()
+      list(APPEND _lst "-l${item}")
+    endif()
```

Comment:
I've partially ported commit  6e121b2 from branch 2.4. That commit removed the `-L` options appearance for CUDA libraries (other part will be ported soon, I believe).
The '-Wl' option is not possible, I used it for experiments with different linking flags on early stages without success.